src/plugins/external/session.ts

```diff
@@ -0,0 +1,32 @@
+import fastifySession, { FastifySessionOptions } from '@fastify/session'
+import { FastifyInstance } from 'fastify'
+import fp from 'fastify-plugin'
+import { Auth } from '../../schemas/auth.js'
+
+declare module 'fastify' {
+  interface Session {
+    user: Auth
+  }
+}
+
+export const autoConfig = (
+  fastify: FastifyInstance
+): FastifySessionOptions => ({
+  secret: fastify.config.COOKIE_SECRET,
+  cookieName: fastify.config.COOKIE_NAME,
+  cookie: {
+    secure: fastify.config.COOKIE_SECURED,
+    httpOnly: true,
+    maxAge: 1800000 // 30 minutes
+  }
```

Comment:
I'd advice to also make this one configurable

Reply:
Ok